ios/RNSScreen.mm

```diff
@@ -571,31 +612,33 @@ - (void)updatePresentationStyle
       } else {
         RCTLogError(@"Unhandled value of sheetLargestUndimmedDetent passed");
       }
-
-      if (_sheetAllowedDetents == RNSScreenDetentTypeMedium) {
-        sheet.detents = @[ UISheetPresentationControllerDetent.mediumDetent ];
-        if (sheet.selectedDetentIdentifier != UISheetPresentationControllerDetentIdentifierMedium) {
-          [sheet animateChanges:^{
-            sheet.selectedDetentIdentifier = UISheetPresentationControllerDetentIdentifierMedium;
-          }];
-        }
-      } else if (_sheetAllowedDetents == RNSScreenDetentTypeLarge) {
-        sheet.detents = @[ UISheetPresentationControllerDetent.largeDetent ];
-        if (sheet.selectedDetentIdentifier != UISheetPresentationControllerDetentIdentifierLarge) {
-          [sheet animateChanges:^{
-            sheet.selectedDetentIdentifier = UISheetPresentationControllerDetentIdentifierLarge;
-          }];
-        }
-      } else if (_sheetAllowedDetents == RNSScreenDetentTypeAll) {
-        sheet.detents =
-            @[ UISheetPresentationControllerDetent.mediumDetent, UISheetPresentationControllerDetent.largeDetent ];
-      } else {
-        RCTLogError(@"Unhandled value of sheetAllowedDetents passed");
-      }
     }
   }
-#endif // Check for max allowed iOS version
+#endif // Check for iOS >= 15
 }
+
+#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && defined(__IPHONE_16_0) && \
+    __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_16_0
+- (NSArray<UISheetPresentationControllerDetent *> *)detentsFromMaxHeightFractions:(NSArray<NSNumber *> *)fractions
+    API_AVAILABLE(ios(16.0))
+{
+  NSMutableArray<UISheetPresentationControllerDetent *> *customDetents =
+      [NSMutableArray arrayWithCapacity:fractions.count];
+  int detentIndex = 0;
+  for (NSNumber *frac in fractions) {
+    NSString *ident = [[NSNumber numberWithInt:detentIndex] stringValue];
```

Comment:
System API is organised is such way, that if I want to refer to particular detent I need to do it by its identifier (unique string). Thus I'm using indexes here. I rejected doing it by detent value (height) because the value is context dependent.

Alternative would be to make user define detents as objects with `id` / `name` field, e.g.:

```js
[{
  id: "3/5",
  value: 0.6
}]
```

But I did not like this idea even more so. 


This is why `sheetCustomLargestUndimmedDetent` (the name must be changed) takes an index as a value.

Reply:
What about if someone provided 2 same values?

Reply:
iOS will treat them as a single detent. The resolved value in such case is rather iOS implementation specific and not documented. It's better to treat such case as undefined behaviour. 

I see your point though.

I guess we will stick to using indices.